src/main/java/io/excaliburfrc/robot/subsystems/Shooter.java

```diff
@@ -13,12 +18,59 @@
       new CANSparkMax(ShooterConstants.LEADER_ID, MotorType.kBrushless);
   private final CANSparkMax follower =
       new CANSparkMax(ShooterConstants.FOLLOWER_ID, MotorType.kBrushless);
+  private final Encoder encoder =
+      new Encoder(ShooterConstants.ENCODER_A, ShooterConstants.ENCODER_B);
+  private final SimpleMotorFeedforward feedforward =
+      new SimpleMotorFeedforward(ShooterConstants.kS, ShooterConstants.kV);
+  private final PIDController pid =
+      new PIDController(ShooterConstants.kP, ShooterConstants.kI, ShooterConstants.kD);
+
+  private boolean isControl = false;
 
   public Shooter() {
     follower.follow(leader);
   }
 
-  public Command activateCommand(DoubleSupplier speed) {
+  public Command manualCommand(DoubleSupplier speed) {
     return new RunCommand(() -> leader.set(speed.getAsDouble() * 0.5), this);
   }
+
+  public Command accelerateToVelocityCommand(double rpm) {
+    return new StartEndCommand(() -> accelerate(rpm), this::release);
+  }
+
+  private void accelerate(double setpoint) {
+    pid.setSetpoint(setpoint);
+    isControl = true;
+  }
+
+  private void release() {
+    pid.setSetpoint(0); // for safety reasons
+    isControl = false;
+  }
+
+  private double prev_x = 0;
+  private double prev_t = Timer.getFPGATimestamp() - 0.02;
+
+  public double getVelocity() {
+    double _prevX = prev_x;
+    double _prevT = prev_t;
+
+    prev_x = encoder.getDistance();
+    prev_t = Timer.getFPGATimestamp();
+
+    double dx = prev_x - _prevX;
+    double dt = prev_t - _prevT;
+
+    return dx / dt;
+  }
+
+  @Override
+  public void periodic() {
+    if (isControl) {
+      double ffOutput = feedforward.calculate(pid.getSetpoint());
+      double pidOutput = pid.calculate(getVelocity());
+      leader.set(pidOutput + ffOutput);
+    }
```

Comment:
And what happens when control is released?

Reply:
`private void release() {
    pid.setSetpoint(0); // for safety reasons
    isControl = false;
  }`

Reply:
Yes, but what does the motor output?

Reply:
Should it wait until the velocity reaches 0 and then do isControl = false?

Reply:
Running control on 0 RPM isn't a good idea, we've talked about this.
If we're not running control, we want to apply 0 volts to the motor: `leader.set(0)`.

Reply:
Doesn't that happen automatically when there is no output?

Reply:
No. `MotorSafety` exists, but we don't want to count on it.

Reply:
ok.
does the `pid.setSetpoint(0);` do anything if it's just disabled instantly after that?